src/ui/flutter_app/lib/services/mill/src/tap_handler.dart

```diff
@@ -84,53 +81,41 @@ class TapHandler {
             ret = true;
             logger.v("[tap] selectPiece: [$sq]");
 
-            final us = controller.gameInstance.sideToMove;
+            final pieceOnBoardCount =
+                position.pieceOnBoardCount[controller.gameInstance.sideToMove];
             if (position.phase == Phase.moving &&
                 DB().rules.mayFly &&
-                (controller.position.pieceOnBoardCount[us] ==
-                        DB().rules.flyPieceCount ||
-                    controller.position.pieceOnBoardCount[us] == 3)) {
+                (pieceOnBoardCount == DB().rules.flyPieceCount ||
+                    pieceOnBoardCount == 3)) {
+              // TODO: [Calcitem, Leptopoda] why is the [DB().rules.flyPieceCount] not respected?
```

Comment:
In this case a less than operation might make the code more readable.

Reply:
Yes!

Reply:
I remembered one thing. If it is modified, it needs to be verified whether it can be handled correctly when the number of stones is less than 3. It is expected that the game is over and not here.  

By the way, much of the logic of the position module still uses the basic logic of NineChess. There are many if and else branches. I don't know if there is a better way to re-implement it, perhaps using a state machine?  

However, the reconstruction of these functions does require careful testing with test cases, and the workload is relatively large.